File: 2015/python/aoc2.py
input_path = '../inputs/day_2.txt'

x = open(input_path).read()
lines = x.split('\n')

def area(s):
    d = [int(x) for x in  s.split('x')]
    a = 2*d[0]*d[1]+2*d[1]*d[2]+2*d[0]*d[2]
    d.remove(max(d))
    a = a + d[0]*d[1]
    return a

def length(s):
    d = [int(x) for x in  s.split('x')]
    v = d[0]*d[1]*d[2]
    d.remove(max(d))
    p = 2*d[0]+2*d[1] + v
    return p

#part 1
a = 0
for l in lines:
    a += area(l)

print(a)

#part 2
p = 0
for l in lines:
    p += length(l)

print(p)

File: 2023/matlab/aoc2.m
input_path = fullfile(get_input_root,'day_2.txt');

x = read_txt(input_path);
x = strip(x);
lines = strsplit(x,newline);

g = cell(numel(lines),1);

%part1 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
max_vals = zeros(numel(lines),3);
for i1 = 1:numel(lines)
    line_i1 = lines{i1};
    reveals = strsplit(extractAfter(line_i1,':'),';');
    r = cell(numel(reveals),1);
    m = containers.Map({'red','green','blue'},[0,0,0]);
    for i2 = 1:numel(reveals)
        sets = strsplit(reveals{i2},',');
        s = cell(numel(sets),1);
        for i3 =  1:numel(sets)
            s{i3} = strsplit(strip(sets{i3}),' ');
            ni3 = str2double(s{i3}{1});
            m(s{i3}{2}) = max(m(s{i3}{2}),ni3 );
        end
        r{i2} = s;
    end
    v  = m.values;
    max_vals(i1,:) =[v{:}];
    g{i1} = r;
end
check = [14,13,12];

p1 = sum(find(~any(max_vals > check,2)));
disp(p1)

%part2 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
sum(max_vals(:,1).*max_vals(:,2).*max_vals(:,3))

